src/autoscaler/scalingengine/schedule/schedules.go

```diff
@@ -0,0 +1,104 @@
+package schedule
+
+import (
+	"autoscaler/cf"
+	"autoscaler/db"
+
+	"code.cloudfoundry.org/lager"
+	"sync"
+)
+
+type ActiveSchedule struct {
+	ScheduleId      string
+	InstanceMin     int `json:"instance_min_count"`
+	InstanceMax     int `json:"instance_max_count"`
+	InstanceInitial int `json:"initial_min_instance_count"`
+}
+
+type AppSchedules struct {
+	schedules map[string]ActiveSchedule
+	lock      *sync.Mutex
+	cfClient  cf.CfClient
+	logger    lager.Logger
+	policyDB  db.PolicyDB
+}
+
+func NewAppSchedules(logger lager.Logger, cfClient cf.CfClient, policyDB db.PolicyDB) *AppSchedules {
+	return &AppSchedules{
+		schedules: make(map[string]ActiveSchedule),
+		lock:      &sync.Mutex{},
+		cfClient:  cfClient,
+		policyDB:  policyDB,
+		logger:    logger,
+	}
+}
+
+func (as *AppSchedules) GetActiveSchedule(appId string) (ActiveSchedule, bool) {
+	as.lock.Lock()
+	schedule, exists := as.schedules[appId]
+	as.lock.Unlock()
+	return schedule, exists
+}
+
+func (as *AppSchedules) SetActiveSchedule(appId string, schedule ActiveSchedule) error {
+	as.lock.Lock()
+	currentSchedule, exists := as.schedules[appId]
+	as.schedules[appId] = schedule
+	as.lock.Unlock()
+
+	logger := as.logger.WithData(lager.Data{"appId": appId})
+
+	if exists {
+		logger.Info("set-active-schedule", lager.Data{"message": "active schedule exists", "currentSchedule": currentSchedule})
+	}
+
+	err := as.cfClient.SetAppInstances(appId, schedule.InstanceInitial)
```

Comment:
Is InstanceInitial optional?  We may need to think about how this is used.
If you are below the min, you should get the initialmin.
If you are above max, you should go to max?
if you are above initialMin, you should stay where you are.